lib/improver/utilities/cube_extraction.py

```diff
@@ -129,9 +157,11 @@ def parse_constraint_list(constraints, units=None):
             complex_constraints.append(create_range_constraint(key, value))
         else:
             try:
-                simple_constraints_dict[key] = literal_eval(value)
+                typed_value = literal_eval(value)
             except ValueError:
                 simple_constraints_dict[key] = value
+            else:
```

Comment:
Just as a query, I don't think that we actually have a unit test that shows the usage of the `ValueError` exception above.  I get the impression e.g. from [this webpage](https://bugs.python.org/issue31778) that this is for input such as `threshold=2+2`. Perhaps this should have a unit test to show what the intended usage of this `ValueError` is?

Reply:
Added a string constraint test to the Test_parse_constraint_list class which goes via this route.